lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -94,7 +95,7 @@ export default class BitcoinProcessor {
   /** at least 10 blocks per page unless reaching the last block */
   private static readonly pageSizeInBlocks = 10;
 
-  public constructor (config: IBitcoinConfig) {
+  public constructor (config: IBitcoinConfig, bitcoinWallet?: IBitcoinWallet) {
```

Comment:
Optional: if you took the earlier suggestion, you might want to make the signature the same here, and remove it from IBitcoinConfig.
```suggestion
  public constructor (config: IBitcoinConfig, bitcoinWalletOrImportString: IBitcoinWallet | string) {
```

Reply:
Updated to pass in the bitcoinwallet as part of the config.